lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -590,9 +590,11 @@ def get(self, request, course_id=None):
         threads_per_page = request.GET.get('page_size', 10)
         discussion_id = None
         try:
-            user_id = int(request.GET.get('user_id', None))
-        except (TypeError, ValueError):
-            return Response({'details': 'Invalid user id'}, status=400)
+            username = request.GET.get('username', None)
+            user = User.objects.get(username=username)
+            user_id = int(user.id)
+        except (TypeError, ValueError, ObjectDoesNotExist):
```

Comment:
What are the cases these 4 errors will occur? Is it possible to remove this try catch?